core/src/main/java/io/grpc/Metadata.java

```diff
@@ -642,9 +645,9 @@ private static String validateName(String n) {
       return n;
     }
 
-    private Key(String name) {
+    private Key(String name, boolean pseudo) {
```

Comment:
No for put.  if they cannot create keys that are pseudo , there is no need to check.

User who try to read `:status` may have a legitimate need, but I don't think it is currently supported.  If we changed the transport to inProcess, the result would not make sense.  Any other transport would reveal this problem.  Until we have thought about how to support this, I would like to prevent it.